pkg/query/columnquery_test.go

```diff
@@ -499,74 +500,77 @@ func TestColumnQueryAPIQueryDiff(t *testing.T) {
-	require.NoError(t, err)
-	p, err := profile.Parse(bytes.NewBuffer(fileContent))
-	require.NoError(t, err)
-	profiles, err := parcaprofile.ProfilesFromPprof(ctx, logger, m, p, false)
-	require.NoError(t, err)
-	require.Equal(t, 4, len(profiles))
-	for _, prof := range profiles {
-		_, err = parcacol.InsertProfileIntoTable(ctx, logger, table, labels.Labels{{
-			Name:  "__name__",
-			Value: "memory",
-		}, {
-			Name:  "job",
-			Value: "default",
-		}}, prof)
+	arch := runtime.GOARCH
+	if arch == "amd64" {
+		ctx := context.Background()
+		logger := log.NewNopLogger()
+		reg := prometheus.NewRegistry()
+		tracer := trace.NewNoopTracerProvider().Tracer("")
+		col := columnstore.New(
+			reg,
+			8196,
+			64*1024*1024,
+		)
+		colDB, err := col.DB("parca")
+		require.NoError(t, err)
+		table, err := colDB.Table(
+			"stacktraces",
+			columnstore.NewTableConfig(
+				parcacol.Schema(),
+			),
+			logger,
+		)
+		require.NoError(t, err)
+		m := metastore.NewBadgerMetastore(
+			logger,
+			reg,
+			tracer,
+			metastore.NewRandomUUIDGenerator(),
+		)
+		t.Cleanup(func() {
+			m.Close()
+		})
+
+		fileContent, err := ioutil.ReadFile("testdata/alloc_space_delta.pb.gz")
+		require.NoError(t, err)
+		p, err := profile.Parse(bytes.NewBuffer(fileContent))
+		require.NoError(t, err)
+		profiles, err := parcaprofile.ProfilesFromPprof(ctx, logger, m, p, false)
+		require.NoError(t, err)
+		require.Equal(t, 4, len(profiles))
+		for _, prof := range profiles {
+			_, err = parcacol.InsertProfileIntoTable(ctx, logger, table, labels.Labels{{
+				Name:  "__name__",
+				Value: "memory",
+			}, {
+				Name:  "job",
+				Value: "default",
+			}}, prof)
+			require.NoError(t, err)
+		}
+
+		table.Sync()
+
+		api := NewColumnQueryAPI(
+			logger,
+			tracer,
+			m,
+			query.NewEngine(
+				memory.DefaultAllocator,
+				colDB.TableProvider(),
+			),
+			"stacktraces",
+		)
+		res, err := api.ProfileTypes(ctx, &pb.ProfileTypesRequest{})
 		require.NoError(t, err)
-	}
-
-	table.Sync()
-
-	api := NewColumnQueryAPI(
-		logger,
-		tracer,
-		m,
-		query.NewEngine(
-			memory.DefaultAllocator,
-			colDB.TableProvider(),
-		),
-		"stacktraces",
-	)
-	res, err := api.ProfileTypes(ctx, &pb.ProfileTypesRequest{})
-	require.NoError(t, err)
 
-	require.True(t, proto.Equal(&pb.ProfileTypesResponse{Types: []*pb.ProfileType{
-		{Name: "memory", SampleType: "alloc_objects", SampleUnit: "count", PeriodType: "space", PeriodUnit: "bytes", Delta: true},
-		{Name: "memory", SampleType: "alloc_space", SampleUnit: "bytes", PeriodType: "space", PeriodUnit: "bytes", Delta: true},
-		{Name: "memory", SampleType: "inuse_objects", SampleUnit: "count", PeriodType: "space", PeriodUnit: "bytes", Delta: true},
-		{Name: "memory", SampleType: "inuse_space", SampleUnit: "bytes", PeriodType: "space", PeriodUnit: "bytes", Delta: true},
-	}}, res))
+		require.True(t, proto.Equal(&pb.ProfileTypesResponse{Types: []*pb.ProfileType{
```

Comment:
This one is quite falky. `res` returned in arm machine is not as the same order everytime. The ordering on `SampleType: "inuse_objects"` and `SampleType: "inuse_space"` is interchangeable and different every time the test is run. Figuring out a way to fix this.